File: src/RNetPi.Core/Packets/PacketS2CZoneDeleted.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x05
/// Zone Deleted
/// Informs client of a deleted zone
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
/// </summary>
public class PacketS2CZoneDeleted : PacketS2C
{
    public const byte ID = 0x05;

    public PacketS2CZoneDeleted(byte controllerID, byte zoneID)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
    }

    public override byte GetID() => ID;
}

File: src_js/server/packets/PacketS2CZoneDeleted.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x05
 * Zone Deleted
 * Informs client of a deleted zone
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 */
class PacketS2CZoneDeleted extends PacketS2C {
    constructor(ctrlrID, zoneID) {
        super();

        this._buffer.writeUInt8(ctrlrID);
        this._buffer.writeUInt8(zoneID);
    }

    getID() {
        return 0x05;
    }
}

module.exports = PacketS2CZoneDeleted;
